pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -996,6 +1055,23 @@ public void addUnAckedMessages(int numberOfMessages) {
         topic.getBrokerService().addUnAckedMessages(this, numberOfMessages);
     }
 
+    @Override
+    public void afterAckMessages(Throwable exOfDeletion, Object ctxOfDeletion) {
+        /**
+         * - Mark an acknowledgement were executed.
+         * - If there was no previous pause due to cursor data is too large to persist, we don't need to manually
+         *   trigger a new read. This can avoid too many CPU circles.
+         * - Clear the marker that represent delivery was paused at least once in the earlier time.
+         */
+        blockSignatureOnCursorDataCanNotFullyPersist.markNewAcknowledged();
+        if (blockSignatureOnCursorDataCanNotFullyPersist.hasPausedAtLeastOnce()
+                && cursor.isCursorDataFullyPersistable()) {
+            // clear paused count, and trigger a new reading.
+            blockSignatureOnCursorDataCanNotFullyPersist.clearMarkerPaused();
+            readMoreEntriesAsync();
+        }
+    }
```

Comment:
I tried another approach to simplify the implementation.

AtomicBoolean pausedByCursorDataPersistence;

```java
public void afterAckMessages(Throwable exOfDeletion, Object ctxOfDeletion) {
        if (pausedByCursorDataPersistence.get()) {
	    if (cursor.isCursorDataFullyPersistable()) {
	    	if (pausedByCursorDataPersistence.compareAndSet(true, false)) {
	    		readMoreEntriesAsync();
	    	} else {
	    	    afterAckMessages(exOfDeletion, ctxOfDeletion)
	    	}
	    }
	} else {
		if (!cursor.isCursorDataFullyPersistable()) {
		    if (pausedByCursorDataPersistence.compareAndSet(false, true)) {

		    } else {
		    	afterAckMessages(exOfDeletion, ctxOfDeletion)
		    }
		}
	}
}
```

Reply:
Good suggestion, Improved. Please take a look